extensions/database/src/main/java/com/google/android/agera/database/SqlRequestCompiler.java

```diff
@@ -112,16 +121,40 @@ public Object arguments(@NonNull final String... arguments) {
 
```

Comment:
The compiler never "switches back", since it's building a single request only. (although the built request could technically be reused). For new requests with changed algorithm a new needs to be _compiled_